pkg/kube/configmapwatcher/configmapwatcher.go

```diff
@@ -131,7 +131,27 @@ func (c *Controller) processNextWorkItem() bool {
 	defer c.queue.Done(obj)
 
 	log.Debug("processing queue item")
-	c.callback()
-	c.queue.Forget(obj)
+	if err := c.processItem(); err != nil {
+		log.Error("error processing queue item (retrying)")
+		c.queue.AddRateLimited(struct{}{})
+	} else {
+		c.queue.Forget(obj)
+	}
 	return true
 }
+
+func (c *Controller) processItem() error {
+	cm, err := c.informer.Lister().ConfigMaps(c.configMapNamespace).Get(c.configMapName)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("error fetching object %s error: %v", c.configMapName, err)
+		}
+		cm = nil
+	}
+	c.callback(cm)
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.hasSynced = true
```

Comment:
will hasSynced ever be true if there is no mesh config? I am a bit worried we will get "stuck" and not have any indication about what is going on

Reply:
It can be true when there's no ConfigMap in the API server, yeah. HasSynced means that a first sync with the API server has been done. What do you mean by getting stuck?

Reply:
I think with this change, if mesh config is missing, then cache.WaitForCache sync will block forever

Reply:
It doesn't block due to [this](https://github.com/istio/istio/pull/27760/files#diff-691e37262a21390173f949f47b9d0c06R110). The initial test case has no config

Reply:
thanks, I missed that